### PR TITLE
Feature/enable rt1170

### PR DESCRIPTION
**Pull request by Lucien-Zhao**
Update drivers according rt1170 rc2 package.

### CODE REVIEW

File: mcux/CMakeLists.txt [hunk 1]
```diff
@@ -46,6 +52,27 @@ if (${MCUX_DEVICE} MATCHES "LPC|MIMXRT6")
   zephyr_library_sources(devices/${MCUX_DEVICE}/fsl_reset.c)
 endif()
 
+if (${MCUX_DEVICE} MATCHES "MIMXRT117")
+  zephyr_library_sources(devices/${MCUX_DEVICE}/fsl_romapi.c)
```
Comment (galak) [inline mcux/CMakeLists.txt:56]: Feels like some of these should be enabled based on driver support (for example fsl_romapi.c seems related to flash nor driver).

**Review by galak — COMMENTED** [review]
@MaureenHelm should we look at moving mcux/boards/* code into the Zephyr tree?

**Comment by MaureenHelm** [comment]
> @MaureenHelm should we look at moving mcux/boards/* code into the Zephyr tree?

Why?

File: mcux/CMakeLists.txt (hunk 1 above)
Comment (Lucien-Zhao) [reply mcux/CMakeLists.txt:56]: ROMAPI is not a common driver file. The API functions of different soc are different, but in a specific soc we call the fsl_romapi file that provide api function.

Comment (galak) [reply mcux/CMakeLists.txt:56]: Do we need any special linker stuff for romapi?

Comment (Lucien-Zhao) [reply mcux/CMakeLists.txt:56]: ROMAPI don't need special link stuff. It is only some API interface(address) available to the user. These functions have been specified in ROM code's address.

File: mcux/drivers/imx/CMakeLists.txt [hunk 2]
```diff
@@ -14,6 +14,7 @@ zephyr_compile_definitions_ifdef(
 zephyr_library_compile_definitions_ifdef(
   CONFIG_HAS_MCUX_CACHE FSL_SDK_ENABLE_DRIVER_CACHE_CONTROL
 )
+zephyr_library_sources(fsl_common.c)
```
Comment (mmahadevan108) [inline mcux/drivers/imx/CMakeLists.txt:18]: Why is this needed?

File: mcux/README [hunk 3]
```diff
@@ -30,6 +30,7 @@ Status:
    MIMXRT1061	SDK 2.8.2 (2020-08-19)
    MIMXRT1062	SDK 2.8.2 (2020-08-19)
    MIMXRT1064	SDK 2.8.2 (2020-08-19)
+   MIMXRT1170   SDK 2.9.0 (2021-01-29)
```
Comment (mmahadevan108) [inline mcux/README:33]: Minot issue, could you please fix the spacing

**Comment by mmahadevan108** [comment]
@Lucien-Zhao any update

**Comment by mmahadevan108** [comment]
@Lucien-Zhao any update?

**Comment by CristXu** [comment]
Hi，
    I try to fetch this PR  to my local branch, but i found that, it said:
image
   But, it seems that there is without a folder named imxrt117x.
   I want to check if this is a typo error?

Regards,
    Crist

**Comment by Lucien-Zhao** [comment]
Hi @mmahadevan108 ,
    Why you only add a file named rt117x into CMAKELIST? However, there is no file named rt117x.

**Comment by mmahadevan108** [comment]
I only squashed 2 commits related to drivers folder. I have not added any new content. I will update to remove this. Thanks for pointing it out.

File: mcux/drivers/imx/CMakeLists.txt (hunk 2 above)
Comment (Lucien-Zhao) [reply mcux/drivers/imx/CMakeLists.txt:18]: In fact, fsl_common.c/h provide developer  many common functions to use. In our sdk, all rt series project will include this file.  On RT1170 board, this file must be  compiled into image because PMU unit need delay function to ensure power of PLL unit stable.

File: mcux/README (hunk 3 above)
Comment (Lucien-Zhao) [reply mcux/README:33]: However I don't update device information for other RT soc. I can't update readme for other rt soc because I update the common drivers.

**Comment by CristXu** [comment]
Hi，
   I found another potential bug inside the modules/hal/nxp/mcux/CMakeLists.txt:
image
   We need to use the MCUX_DEVICE as MIMXRT1176_CM7 / MIMXRT1176_CM4 at line 65/72, but the code set the MCUX_DEVICE as a new one: MIMXRT1176, this will cause that te fsl_cache.c can not be included into the project.

Regards,
    Crist

**Comment by mmahadevan108** [comment]
> Hi，
> I found another potential bug inside the modules/hal/nxp/mcux/CMakeLists.txt:
> image
> We need to use the MCUX_DEVICE as MIMXRT1176_CM7 / MIMXRT1176_CM4 at line 65/72, but the code set the MCUX_DEVICE as a new one: MIMXRT1176, this will cause that te fsl_cache.c can not be included into the project.
> 
> Regards,
> Crist

The cache file for the M7 architecture is already picked up from the drivers folder similar to other RT platforms. This change is only for the M4 fsl_cache.c file.

File: mcux/devices/MK64F12/MK64F12_features.h [hunk 4]
```diff
@@ -392,6 +392,8 @@
 #define FSL_FEATURE_ENET_HAS_RECEIVE_STATUS_THRESHOLD (1)
 /* @brief Has Extend MDIO Support. */
 #define FSL_FEATURE_ENET_HAS_EXTEND_MDIO (0)
+/* @brief Queue Size for each instance. */
```
Comment (galak) [inline mcux/devices/MK64F12/MK64F12_features.h:395]: Should we not just resync these files with SDK 2.9?

Comment (mmahadevan108) [reply mcux/devices/MK64F12/MK64F12_features.h:395]: I feel we can do a separate PR to update all platforms to SDK 2.9

Comment (galak) [reply mcux/devices/MK64F12/MK64F12_features.h:395]: I agree a separate PR, but seems like doing that first might be better and easier to see what's going on and keeping things in sync.  @MaureenHelm thoughts?

Comment (mmahadevan108) [reply mcux/devices/MK64F12/MK64F12_features.h:395]: I am going with this approach for now as there is a request to get RT1170 support into the mainline at the earliest. There is a team waiting to add more features.

Comment (mmahadevan108) [reply mcux/devices/MK64F12/MK64F12_features.h:395]: I will work on a separate PR to update to SDK 2.9

Comment (MaureenHelm) [reply mcux/devices/MK64F12/MK64F12_features.h:395]: Initial RT1170 support has been stuck for a long time, so I'd like to get that unstuck and then come back and do a full SDK 2.9 update for all SoCs afterward.

**Review by MaureenHelm — APPROVED** [review]
